api/jobset/v1alpha2/jobset_webhook_test.go

```diff
@@ -562,3 +564,119 @@ func TestValidateCreate(t *testing.T) {
 		})
 	}
 }
+
+func TestPatchPod(t *testing.T) {
+	var (
+		topologyKey = "topology-key"
+		jobName     = "job"
+		podName     = "pod"
+		ns          = "default"
+	)
+	tests := []struct {
+		name         string
+		pod          *corev1.Pod
+		wantAffinity corev1.Affinity
+		wantGate     *corev1.PodReadinessGate
+	}{
+		{
+			name: "completion index 0",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: podName,
+					Annotations: map[string]string{
+						batchv1.JobCompletionIndexAnnotation: "0",
+						ExclusiveKey:                         topologyKey,
+					},
+					Labels: map[string]string{
+						JobKey: names.JobHashKey(ns, jobName),
```

Comment:
We don't need to use JobHashKey function here (and so we don't need to export it), just set any value manually.